feat: for-expr with variables in generate_hcl.content (#1714)
terramate-io/terramate: CHANGELOG.md

## Unreleased

### Added

- Add support for `[for ...]` and `{for ...}` expressions containing Terramate variables and functions inside the `generate_hcl.content` block.

### Changed

> - (**BREAKING CHANGE**) The code generated was changed the format but still semantically the same as before. This change is marked as "breaking" because can be triggered to be executed after upgrading due to different code generated.

Reviewer: ```suggestion
- (**BREAKING CHANGE**) The format of the generated code may change while being still semantically the same as before. This change is marked as "breaking", because this may trigger change detection on files where the formatting changes formatting.
```
